src/platform/common/utils/multiStepInput.ts

```diff
@@ -146,7 +163,25 @@ export class MultiStepInput<S> implements IMultiStepInput<S> {
                             resolve(<any>item);
                         }
                     }),
-                    input.onDidChangeSelection((selectedItems) => resolve(selectedItems[0])),
+                    input.onDidChangeSelection(async (selectedItems) => {
+                        const itemLabel = selectedItems.length ? selectedItems[0].label : '';
+                        let resolvable = itemLabel ? true : false;
+                        if (itemLabel && validate) {
+                            input.enabled = false;
+                            input.busy = true;
+                            const message = await validate(itemLabel);
+                            if (message) {
+                                resolvable = false;
+                                // No validation allowed on a quick pick. Have to put up a dialog instead
+                                await this.shell.showErrorMessage(message, { modal: true });
```

Comment:
I think I need to change this to support markdown. For remote there's a URL the user can click on and it doesn't seem to be clickable at the moment. Maybe dialogs don't support this.

Reply:
I think they don't sadly. Our "Install Python" popup tries to link to Python.org, but it's not clickable. Pretty sure when we talked about it in v-team that's just how it is.